### ISSUE TITLE
Error in tensorflow_probability.optimizer

### ISSUE THREAD

**Issue by Karthikprabhu22** [tensorflow/probability#653]
**System information**

OS Platform and Distribution: Ubuntu 18.04.3 LTS
TensorFlow installed from : pip install from anaconda
TensorFlow version : ''2.0.0''
TensorFlow Probability version : '0.8.0-rc0'
Python version: 3.7.3

**Describe the current behavior**
When trying to use the tfp.optimizer.lbfgs_minimize function, I get an error :
> 'then' must be at least a vector, but saw shape: [] [Op:Select]

**Relevant part of the code:**
`"""generator is a previously trained model. map_vector is tensor with the same shape as the output of the generator. 
"""

x = tf.random.normal((1,64))

def l2(x):
    with tf.GradientTape() as tape:
        tape.watch(x)
        loss_value = tf.linalg.norm(generator(x) - map_vector)
    jac = tape.gradient(loss_value,x)
    return loss_value, jac

opt = tfp.optimizer.lbfgs_minimize(l2,initial_position = x, tolerance=1e-5)
`

**Comment by Karthikprabhu22** [comment]
The issue has been resolved.
The output of l2(x) was:

> <tf.Tensor: id=1615, shape=(), dtype=float32, numpy=7.149333>,
 <tf.Tensor: id=1681, shape=(1, 64), dtype=float32, numpy=
 array([[ 1.0233184 , -0.67168254, -0.75814754,  2.8783326 ,  1.3204312 ,
          0.4518584 ,  3.76576   ,  1.0825894 ,  0.50309736,  1.2092152 ,
         -2.0877633 , -0.26971212, -2.1024365 , -0.07273053, -0.22633885,
          0.7087722 ,  1.0272506 ,  0.4062516 , -0.5084653 , -0.22504732,
         -0.87500036,  0.22659774, -1.3053032 , -3.210107  ,  2.5480978 ,
          2.2262604 ,  0.12413362, -0.6265972 ,  1.259634  ,  1.840487  ,
          1.2816948 ,  0.21707988,  0.94418347,  1.5250891 , -0.939271  ,
         -4.116663  ,  3.2982059 ,  1.1132284 , -3.9772592 ,  0.7469107 ,
         -0.20234983,  1.062207  ,  3.3796198 ,  0.2898029 ,  1.4736251 ,
         -1.288457  , -1.7015343 , -1.734964  , -0.03626216, -0.87840134,
          0.8423059 , -0.0730313 , -3.0173016 ,  0.20565715,  2.7258925 ,
         -0.16882756,  3.126473  , -4.31707   ,  1.7206887 , -1.2390788 ,
         -0.18431555,  1.9576714 , -0.23630224, -0.67891705]],
       dtype=float32)>)

The shape of the loss function was (). After adding a line in the definition of the function to reshape it, the optimization worked fine.
